packages/adapter-fauna/src/index.ts

```diff
@@ -10,140 +9,68 @@
 } from "@auth/core/adapters"
 
-export const collections = {
-  Users: Collection("users"),
-  Accounts: Collection("accounts"),
-  Sessions: Collection("sessions"),
-  VerificationTokens: Collection("verification_tokens"),
-} as const
-
-export const indexes = {
-  AccountByProviderAndProviderAccountId: Index(
-    "account_by_provider_and_provider_account_id"
-  ),
-  UserByEmail: Index("user_by_email"),
-  SessionByToken: Index("session_by_session_token"),
-  VerificationTokenByIdentifierAndToken: Index(
-    "verification_token_by_identifier_and_token"
-  ),
-  SessionsByUser: Index("sessions_by_user_id"),
-  AccountsByUser: Index("accounts_by_user_id"),
-} as const
-
-export const format = {
-  /** Takes a plain old JavaScript object and turns it into a Fauna object */
-  to(object: Record<string, any>) {
-    const newObject: Record<string, unknown> = {}
-    for (const key in object) {
-      const value = object[key]
-      if (value instanceof Date) {
-        newObject[key] = Time(value.toISOString())
-      } else {
-        newObject[key] = value
-      }
-    }
-    return newObject
-  },
-  /** Takes a Fauna object and returns a plain old JavaScript object */
-  from<T = Record<string, unknown>>(object: Record<string, any>): T {
-    const newObject: Record<string, unknown> = {}
-    for (const key in object) {
-      const value = object[key]
-      if (value?.value && typeof value.value === "string") {
-        newObject[key] = new Date(value.value)
-      } else {
-        newObject[key] = value
-      }
-    }
-    return newObject as T
-  },
+type ToFauna<T> = {
+  [P in keyof T]: T[P] extends Date | null ? TimeStub | null : T[P]
 }
 
-/**
- * Fauna throws an error when something is not found in the db,
- * `next-auth` expects `null` to be returned
- */
-export function query(f: FaunaClient, format: (...args: any) => any) {
-  return async function <T>(expr: ExprArg): Promise<T | null> {
-    try {
-      const result = await f.query<{
-        data: T
-        ref: { id: string }
-      } | null>(expr)
-      if (!result) return null
-      return format({ ...result.data, id: result.ref.id })
-    } catch (error) {
-      if ((error as errors.FaunaError).name === "NotFound") return null
-      if (
-        (error as errors.FaunaError).description?.includes(
-          "Number or numeric String expected"
-        )
-      )
-        return null
+export type FaunaUser = ToFauna<AdapterUser>
+export type FaunaSession = ToFauna<AdapterSession>
+export type FaunaVerificationToken = ToFauna<VerificationToken>
+export type FaunaAccount = ToFauna<AdapterAccount> & any // TODO: Remove `& any`
 
-      if (process.env.NODE_ENV === "test") console.error(error)
+export const userFields = fql`
+email,
+name,
+image,
+id,
+emailVerified
+`
 
-      throw error
-    }
-  }
-}
+export const sessionFields = fql`
+sessionToken,
+userId,
+expires
+`
+
+export const verificationTokenFields = fql`
+identifier,
+expires,
+token
+`
```

Comment:
This should not be hardcoded. Any ideas on how to achieve this? The fields should be driven by the object keys of the argument of the given adapter method. Maybe `Object.keys(obj).join(",\n")` or something similar can help here.

Reply:
Fixed in https://github.com/nextauthjs/next-auth/pull/9849/commits/93fb4531ffe021215b4f4236ad03c70647d9e6ef